cityofaustin/atd-moped
# Disable save button if required fields are not populated

## Associated issues
Closes https://github.com/cityofaustin/atd-data-tech/issues/20084

This actually ended up being kinda weird to implement, i had to really dig thru github and stack overflow before i found out how to get the edited values while still in edit mode and cause a re-render to enable the save button. Finally I found this github issue where someone had the same problem and a rly helpful [comment](https://github.com/mui/mui-x/issues/11218#issuecomment-1832762046) describing how to handle our use case with [useGridSelector](https://mui.com/x/react-data-grid/state/#access-the-state)

## Testing
**URL to test:** 

https://deploy-preview-1505--atd-moped-main.netlify.app/moped/

**Steps to test:**
1. On the project summary page, test adding a new subproject. The save button should be disabled until you select a project. Make sure the cancel and delete buttons work as well.
2. Test all the other datagrid tables too, Milestones, Funding, and Files. If you dont have the required fields filled out then the save should be disabled. BTW Funding has no required fields.
3. Test editing and deleting required fields, the save button should get disabled again.


---
#### Ship list
   - Added `Try to save a row without the Milestone column populated. You should see the Required helper text and input highlighted in red. The save button should also be disabled.` as a Milestones table test.
